File: javascript/prime-factors/prime-factors.js
module.exports = {
  for: function (limit) {
    var factors = [];
    for (var div = 2; div <= limit; div++)
    {
      while (limit % div == 0)
      {
        factors.push(div);
        limit = limit / div;
      }
    }
    return factors;
  }
};

File: csharp/prime-factors/PrimeFactors.cs
using System.Collections.Generic;

class PrimeFactors
{
  public static int[] For(long number)
  {
    var factors = new List<int>();
    for (var div = 2; div <= number; div++)
    {
      while (number % div == 0)
      {
        factors.Add(div);
        number = number / div;
      }
    }
    return factors.ToArray();
  }
}